### PR TITLE
Fixed rows count in the migration script

### PR DESCRIPTION
**Pull request by alartum**
### Changes:
- replaced `.count()` method with an already existing `get_query_count` method;

### Motivation:

While performing the Apache Airflow update from 2.2.3 to 2.7.1 I faced an issue with the migration:
```bash
$ airflow db migrate
...
File "/home/airflow/.local/bin/airflow", line 8, in <module>
    sys.exit(main())
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/__main__.py", line 60, in main
    args.func(args)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/db_command.py", line 65, in upgradedb
    migratedb(args)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/cli.py", line 113, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/providers_configuration_loader.py", line 56, in wrapped_function
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/db_command.py", line 104, in migratedb
    db.upgradedb(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 77, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/db.py", line 1616, in upgradedb
    for err in _check_migration_errors(session=session):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/db.py", line 1499, in _check_migration_errors
    yield from check_fn(session=session)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/db.py", line 1460, in check_bad_references
    invalid_row_count = bad_rows_query.count()
AttributeError: 'Select' object has no attribute 'count'
```

The cause of the error is that SQLAlchemy [changed the behavior](https://docs.sqlalchemy.org/en/20/changelog/migration_20.html) of the `select` statement. It has already been addressed in other places using `func.count()`, so the problematic line looks more like a typo/oversight.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/utils/db.py [hunk 1]
```diff
@@ -1446,7 +1446,7 @@ class BadReferenceConfig:
 
         dangling_table_name = _format_airflow_moved_table_name(source_table.name, change_version, "dangling")
         if dangling_table_name in existing_table_names:
-            invalid_row_count = bad_rows_query.count()
+            invalid_row_count = get_query_count(bad_rows_query, session=session)
```
Comment (ephraimbuddy) [inline airflow/utils/db.py:1449]: Can we add a test to protect future changes here?

Comment (alartum) [reply airflow/utils/db.py:1449]: Sure, I will add it

Comment (alartum) [reply airflow/utils/db.py:1449]: @ephraimbuddy I've finally found the time to add the test, although it required more work than expected :)

**Review by ephraimbuddy — APPROVED** [review]

File: newsfragments/34348.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixed ``AttributeError: 'Select' object has no attribute 'count'`` during the ``airflow db migrate`` command
```
Comment (ephraimbuddy) [inline newsfragments/34348.bugfix.rst:1]: Nice to have but not required

**Pull request merged**